### PR TITLE
GitBranch fails with empty repos #643

### LINKED ISSUE #643: GitBranch fails with empty repos
I set this up a few days ago https://github.com/cs3217-ay1819s2-instructors/dashboard

Started failing today (on Travis)

>04:03:19 - Log folder has been successfully created
04:03:19 - Cloning in parallel from https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git...
04:03:19 - Waiting for cloning of https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git to complete...
04:03:23 - Cloning of https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git completed!
04:03:23 - Cloning in parallel from https://github.com/cs3217-1819/final-project-cs3217-name.git...
04:03:23 - Analyzing https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master)...
04:03:24 - https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master) does not contain a standalone config file.
04:03:24 - https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master) has no authors specified, using all authors by default.
04:03:24 - Extracting commits info for https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master)...
04:03:24 - Analyzing commits info for https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master)...
04:03:24 - Extracting relevant file info from https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master)...
04:03:31 - Analysis of https://github.com/cs3217-1819/final-project-cs3217-gooreatrestaurant.git (master) completed!
04:03:31 - Waiting for cloning of https://github.com/cs3217-1819/final-project-cs3217-name.git to complete...
04:03:31 - Cloning of https://github.com/cs3217-1819/final-project-cs3217-name.git completed!
04:03:31 - Cloning in parallel from https://github.com/cs3217-1819/final-project-dash.git...
04:03:31 - Analyzing https://github.com/cs3217-1819/final-project-cs3217-name.git (master)...
04:03:31 - https://github.com/cs3217-1819/final-project-cs3217-name.git (master) does not contain a standalone config file.
04:03:31 - https://github.com/cs3217-1819/final-project-cs3217-name.git (master) has no authors specified, using all authors by default.
04:03:31 - Extracting commits info for https://github.com/cs3217-1819/final-project-cs3217-name.git (master)...
04:03:32 - Analyzing commits info for https://github.com/cs3217-1819/final-project-cs3217-name.git (master)...
04:03:32 - Extracting relevant file info from https://github.com/cs3217-1819/final-project-cs3217-name.git (master)...
04:03:33 - Analysis of https://github.com/cs3217-1819/final-project-cs3217-name.git (master) completed!
04:03:33 - Waiting for cloning of https://github.com/cs3217-1819/final-project-dash.git to complete...
04:03:33 - Cloning of https://github.com/cs3217-1819/final-project-dash.git completed!
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at reposense.git.GitBranch.getCurrentBranch(GitBranch.java:23)
	at reposense.report.RepoCloner.getClonedRepoLocation(RepoCloner.java:69)
	at reposense.report.ReportGenerator.cloneAndAnalyzeRepos(ReportGenerator.java:110)
	at reposense.report.ReportGenerator.generateReposReport(ReportGenerator.java:71)
	at reposense.RepoSense.main(RepoSense.java:59)

### PR DESCRIPTION
**Pull request by myteo**
Fixes #643.

```
When analyzing an empty repo, GitBranch throws an uncaught exception
which crashes the program.

This happens because git branch command returns an empty string but
we attempt to split the return value and get the second element
causing ArrayIndexOutOfBoundsException.

Let's catch this uncaught exception and add a unit test.
```

### CODE REVIEW

**Comment by eugenepeh** [comment]
can use https://github.com/reposense/testrepo-Empty

**Comment by myteo** [comment]
@reposense/devs Ready for review.

**Comment by myteo** [comment]
> In case we miss checking that out, and to improve the coverage, perhaps add a unit test?

Added unit test.

File: src/main/java/reposense/report/RepoCloner.java [hunk 1]
```diff
@@ -66,7 +68,15 @@ public RepoLocation getClonedRepoLocation(String outputPath) throws IOException
             return null;
         }
 
-        currentRepoDefaultBranch = GitBranch.getCurrentBranch(configs[currentIndex].getRepoRoot());
+        try {
+            currentRepoDefaultBranch = GitBranch.getCurrentBranch(configs[currentIndex].getRepoRoot());
```
Comment (eugenepeh) [inline src/main/java/reposense/report/RepoCloner.java]: Actually on a second look, this is not a very good solution.
Your proposed message mentioned that GitBranch is the cause of the crash, but the solution is applied on the RepoCloner. This means that any subsequent use of the `GitBranch.getCurrentBranch` elsewhere would still result in a crash, making it an unsafe api.

Can you tweak the solution? Probably make the `GitBranch.getCurrentBranch` to throw a checked exception instead to enforce the use of this API handle the exception.

**Comment by eugenepeh** [comment]
nits to proposed message :

```
The application crashes when the analysis involves an empty
repository.

The causation of this is the GitBranch#getCurrentBranch method; it
splits the value returned by the `git branch`, which can be an empty
string when performed on an empty repository, and attempt to access the
second element of it which raise the ArrayIndexOutOfBoundsException.

Let's fix this by returning a checked exception to ensure users of this
api handles this exception.
```

**Pull request merged**